flang/lib/Semantics/check-cuda.cpp

```diff
@@ -82,6 +82,10 @@ struct DeviceExprChecker
           }
         }
       }
+      if (sym->owner().IsModule() &&
+          DEREF(sym->owner().symbol()).name() == "__cuda_device_builtins") {
```

Comment:
Consider adding `sym->owner().parent().IsIntrinsicModules() &&` as a further check that this is really the built-in module.